### PR TITLE
fix: Move widget name overlay to WidgetsEditor

### LINKED ISSUE #28304: [Bug]: Anvil - First widget name gets cut off
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2023-10-23 at 7 11 39 PM

### Steps To Reproduce

1. Go to https://ce-28238.dp.appsmith.com/
2. Add a widget to a new page
3. The Widget name for the widget is cut off.

### Public Sample App

_No response_

### Environment

Deploy Preview

### Issue video log

_No response_

### Version

https://ce-28238.dp.appsmith.com/

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
- The first widget name in anvil was being cut-off due to the fact that the DOM grandparent of the widget name overlay canvas was not allowing children to be visible beyond bounds.
- In this PR, the widget name overlay canvas renders in the WidgetsEditor (earlier grandparent now parent) instead of MainContainerWrapper (earlier parent now sibling)
- Also, this PR removes the dependency on refs by the widget name overlay canvas

#### PR fixes following issue(s)
Fixes #28304

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress 

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by prsidhu** [comment]
@riodeuno WidgetNameComponent doesn't disappear on scroll and hence stays visible at an incorrect position, until re-draw is forced by focusing or selecting a widegt.

https://github.com/appsmithorg/appsmith/assets/5424788/8be5ca63-b8b5-455b-bdce-fb5a0de48796

File: app/client/src/layoutSystems/common/WidgetNamesCanvas/WidgetNameConstants.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const WIDGET_NAME_FILL_COLORS = {
 //CSS properties of the wrapper object of the html canvas
 export const widgetNameWrapperStyle: CSSProperties = {
   position: "absolute",
-  top: 0,
```
Comment (marks0351) [inline app/client/src/layoutSystems/common/WidgetNamesCanvas/WidgetNameConstants.ts:41]: can we pls use the widget name height as a constant and use it here.

**Review by marks0351 — CHANGES_REQUESTED** [review]

**Comment by riodeuno** [comment]
@prsidhu The issue was that the scroll parent had changed. I've changed the scroll parent to be the WidgetsEditor.

**Review by prsidhu — APPROVED** [review]

**Pull request merged**